logstash-core/src/main/java/org/logstash/config/ir/graph/Graph.java

```diff
@@ -432,7 +432,7 @@ public boolean isEmpty() {
 
     public String uniqueHash() {
         return Util.digest(this.vertices().
-                filter(v -> !(v instanceof QueueVertex)). // has no metadata
+                filter(v -> !(v instanceof QueueVertex) && !(v instanceof SeparatorVertex)). // has no metadata
```

Comment:
nitpick: perhaps `Vertex` should have a method like `Vertex#has_metadata` so we could filter on behaviour instead of a growing list of exceptions.

Reply:
agree. I'll wait to see what you think of my other comments about the `Vertex` strategy.